bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -553,6 +551,22 @@ public FunctionType(Function type, Optional<Integer> windowSize) {
             };
         }
 
+        public boolean alwaysAccept() {
+            if ((type == Function.DELTA || type == Function.DELTA_PERCENT) && acceptedState == UnDefType.UNDEF) {
```

Comment:
While looking at @lolodomo 's comment, I realised that `alwaysAccept()` should accept `acceptedState == UnDefType.NULL` as well as `UnDefType.NULL`. Or??

Reply:
It's a good point. I didn't anticipate that the binding might be sending UnDefTypes through, but I've realised now that that is an incorrect assumption. This needs to be dealt with, even outside the scope of this PR, so I'll do some tests and submit it in a separate PR.

Reply:
The reason it allows it for `acceptedState == UnDefType.UNDEF` is simply because that is the value that `acceptedState` was initialized with (i.e. no previous input had been accepted).

Adding check for `NULL` doesn't apply _for this reason_ because it is simply isn't the initial value.

What I'm saying above is that, as far as I remember (it has been a while, and I have been working on other things since), is that I haven't quite considered what happens when after sending normal numeric data, the binding then sends UNDEF / NULL. This particular case applies to all checks, not just delta / delta_percent functions, hence why it should be dealt with in a separate PR if found necessary.

Reply:
@jimtng in #18144 this [code](https://github.com/openhab/openhab-addons/blob/7a4a9161520fba94a7f505acc98d82e4d73fcd04/bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java#L814) applies a filter to exclude any prior state that is not a valid QuantityType from the set to be calculated. So to be specific it excludes UndefType etc. And the JUnit tests in that PR also test such exclusion.

It would be simple enough to apply the same logic to DecimalType resp. non DecimalType prior states. => If you wish I can do that in #18144 -- or do you want to open yet another PR for that?

Reply:
> It would be simple enough to apply the same logic to DecimalType resp. non DecimalType prior states. => If you wish I can do that in #18144

That's great if you'd like to do it.



> > I chose to use UnDefType.UNDEF. This was probably a less than ideal choice because it made the code a bit unclear
> 
> Yup. It confused me. So perhaps we should revert that, and use an `Optional` or `null` for the non initialised state.

OK, I'll change it